### ISSUE TITLE
Peertube using deprecated X (twitter) API, Thumbnail not shown in tweets

### ISSUE THREAD

**Issue by TNTBOMBOM** [Chocobozzz/PeerTube#6173]
### Describe the current behavior

As the title say.

### Steps to reproduce

Bug report already discussed here:

https://framacolibri.org/t/thumbnail-for-videos-doesnt-show-up-on-x-twitter/20253

1- Download and install peertube
2- Upload video to it
3- Share the link to twitter

e.g: 

https://crimes.media/w/6sK6exmA5CwtcB9oZuc4ot

### Describe the expected behavior

To have a thumbnail for the videos on X (twitter)

### Additional information

PeerTube instance:
Version: 6.0.2
NodeJS version: v21.5.0

**Comment by Chocobozzz** [comment]
Please correctly fill the template as we can't help without the steps to reproduce. For example I don't have issues with twitter to share, for example, this video: https://framatube.org/w/1JPWRE2LkTWLAE4nMKn52Y

**Comment by TNTBOMBOM** [comment]
> Please correctly fill the template as we can't help without the steps to reproduce. For example I don't have issues with twitter to share, for example, this video: https://framatube.org/w/1JPWRE2LkTWLAE4nMKn52Y

Added more info, if you want more specific info please dont hesitate to ask.

**Comment by Chocobozzz** [comment]
Can you try to disable CSP? (even if it's in report only)

**Comment by TNTBOMBOM** [comment]
> Can you try to disable CSP? (even if it's in report only)

Didn't fix it, but here's how it got fixed:

- I had only enabled TLS 1.3 for the instance, so I needed to enable TLS 1.2 to make it work.
- I had to set it to "player" which it would be thumbnail, since Twitter/X hasn't whitelisted my instance yet, Therefore, the player isn't shown, and we are only allowed to use the thumbnail.

ThX! for the help <f>
